src/flag_gems/runtime/device/device.py

```diff
@@ -0,0 +1,51 @@
+import os
+import subprocess
+
+import torch
+import triton
+
+from .. import backend, error
+
+
+class device_ctx:
+    def __init__(self, vendor_name=None):
+        self.vendor_list = backend.vendors_map.keys()
+        self.device_info = self.get_vendor(vendor_name)
+        self.vendor_name, self.device_name, self.cmd, self.vendor = self.device_info
+
+    def get_vendor(self, vendor_name=None) -> tuple:
+        if vendor_name is not None:
+            return backend.get_vendor_info(vendor_name)
+        vendor_from_evn = self._get_vendor_from_evn()
+        if vendor_from_evn is not None:
+            return backend.get_vendor_info(vendor_from_evn)
+        try:
+            return self._get_vendor_from_lib()
+        except Exception as e:
+            error.PASS(e)
+            return self._get_vendor_from_sys()
+
+    def _get_vendor_from_evn(self):
+        device_from_evn = os.environ.get("GEMS_VENDOR")
+        return None if device_from_evn not in self.vendor_list else device_from_evn
+
+    def _get_vendor_from_sys(self):
```

Comment:
This function simply runs a detection to detect if device is available by try a command and expect return code to be zero. Is this a reasonable way to detect all devices?